File: C++/APE6_9.cpp
#include <iostream>
#include <list>
#include <string>

using namespace std;

int main() {

    list<string> turnos;
    int opcion;

    do {
        cout << "\n=== GESTION DE TURNOS ===" << endl;
        cout << "1. Agregar turno al final" << endl;
        cout << "2. Agregar turno al inicio (urgente)" << endl;
        cout << "3. Atender siguiente turno" << endl;
        cout << "4. Ver todos los turnos" << endl;
        cout << "5. Salir" << endl;
        cout << "Opcion: ";
        cin >> opcion;
        cin.ignore();

        switch (opcion) {

            case 1: {
                // Agregar al final
                string nombre;
                cout << "Nombre del paciente/cliente: ";
                getline(cin, nombre);
                turnos.push_back(nombre);
                cout << nombre << " agregado al final de la cola." << endl;
                break;
            }

            case 2: {
                // Agregar al inicio (urgente)
                string nombre;
                cout << "Nombre del caso urgente: ";
                getline(cin, nombre);
                turnos.push_front(nombre);
                cout << nombre << " agregado al INICIO (urgente)." << endl;
                break;
            }

            case 3:
                // Atender al primero
                if (turnos.empty()) {
                    cout << "No hay turnos pendientes." << endl;
                } else {
                    string atendido = turnos.front();
                    turnos.pop_front();
                    cout << "Atendiendo a: " << atendido << endl;
                    cout << "Turnos restantes: " << turnos.size() << endl;
                }
                break;

            case 4:
                // Mostrar turnos
                if (turnos.empty()) {
                    cout << "No hay turnos en la lista." << endl;
                } else {
                    cout << "Turnos actuales:" << endl;
                    int numero = 1;
                    for (const string& turno : turnos) {
                        cout << numero << ". " << turno << endl;
                        numero++;
                    }
                }
                break;

            case 5:
                cout << "Saliendo del programa" << endl;
                break;

            default:
                cout << "Opcion invalida!" << endl;
                break;
        }

    } while (opcion != 5);

    cout << "Sistema cerrado." << endl;

    return 0;
}

File: JAVA/APE6_9.java
package APE6;

import java.util.LinkedList;
import java.util.Scanner;

public class APE6_9 {
    public static void main(String[] args) {
        LinkedList<String> turnos = new LinkedList<>();
        Scanner sc = new Scanner(System.in);
        int opcion;
 
        do {
            System.out.println("\n=== GESTION DE TURNOS ===");
            System.out.println("1. Agregar turno al final");
            System.out.println("2. Agregar turno al inicio (urgente)");
            System.out.println("3. Atender siguiente turno");
            System.out.println("4. Ver todos los turnos");
            System.out.println("5. Salir");
            System.out.print("Opcion: ");
            opcion = sc.nextInt();
            sc.nextLine();
 
            switch (opcion) {
                case 1:
                // Agregar un turno normal al final de la cola
                System.out.print("Nombre del paciente/cliente: ");
                String nombre = sc.nextLine();
                turnos.addLast(nombre);
                System.out.println(nombre + " agregado al final de la cola.");
                break;

                case 2:
                // Agregar un turno urgente al inicio de la cola
                System.out.print("Nombre del caso urgente: ");
                nombre = sc.nextLine();
                turnos.addFirst(nombre);
                System.out.println(nombre + " agregado al INICIO (urgente).");
                break;

                case 3:
                // Atender al primero de la cola y lo eliminar
                if (turnos.isEmpty()) {
                    System.out.println("No hay turnos pendientes.");
                } else {
                    String atendido = turnos.removeFirst();
                    System.out.println("Atendiendo a: " + atendido);
                    System.out.println("Turnos restantes: " + turnos.size());
                }
                break;
 
                case 4:
                // Mostrar todos los turnos en orden
                if (turnos.isEmpty()) {
                    System.out.println("No hay turnos en la lista.");
                } else {
                    System.out.println("Turnos actuales:");
                    int numero = 1;
                    for (String turno : turnos) {
                        System.out.println(numero + ". " + turno);
                        numero++;
                    }
                }
                break;

                case 5:
                System.out.println("Saliendo del programa");
                break;

                default:
                System.out.println("Opcion Invalida!");
                break;
            }
 
        } while (opcion != 5);
 
        System.out.println("Sistema cerrado.");
        sc.close();
    }
}